application/controllers/OptinController.php

```diff
@@ -54,13 +54,11 @@ function actiontokens($surveyid, $token, $langcode = '')
 
         Yii::app()->setLanguage($sBaseLanguage);
 
-        $aSurveyInfo = getSurveyInfo($iSurveyID, $sBaseLanguage);
-
-        if ($aSurveyInfo == false || !tableExists("{{tokens_{$iSurveyID}}}")) {
+        if (empty($oSurvey) || !$oSurvey->hasTokensTable) {
             throw new CHttpException(404, "This survey does not seem to exist. It may have been deleted or the link you were given is outdated or incorrect.");
         } else {
-            LimeExpressionManager::singleton()->loadTokenInformation($iSurveyID, $token, false);
-            $oToken = Token::model($iSurveyID)->findByAttributes(array('token' => $token));
+            LimeExpressionManager::singleton()->loadTokenInformation($iSurveyID, $sToken, false);
+            $oToken = Token::model($iSurveyID)->findByAttributes(array('token' => $sToken));
 
             if (!isset($oToken)) {
                 $sMessage = gT('You are not a participant in this survey.');
```

Comment:
I like to have a HTML error here, unsure of code, maybe 403 ?

Reply:
Maybe, but I was not looking at that here at all :)
Just there was a sanitized $sToken there that was not used and an unsanitized $token used instead